components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -1444,6 +1451,32 @@ impl Animate for Matrix3D {
     }
 }
 
+impl Animate for Matrix {
+    #[cfg(feature = "servo")]
+    fn animate(&self, other: &Self, procedure: Procedure) -> Result<Self, ()> {
+        let this = Matrix3D::from(*self);
+        let other = Matrix3D::from(*other);
+        let this = MatrixDecomposed2D::from(this);
+        let other = MatrixDecomposed2D::from(other);
+        Ok(Matrix3D::from(this.animate(&other, procedure)?).into_2d()?)
+    }
+
+    #[cfg(feature = "gecko")]
+    fn animate(&self, other: &Self, procedure: Procedure) -> Result<Self, ()> {
```

Comment:
Why the difference between servo and Gecko? At least deserves a comment.

Reply:
Not my decision, check out the stuff for matrix3d. This is the same thing translated into a 2d version.

Gecko doesn't follow the spec, basically, but we have to match it to pass tests.

Reply:
This deserves comments, and links to specs.